feat(ui): hide unfocusable floating windows from window lists (#27984)
neovim/neovim: runtime/doc/windows.txt

editor commands such as :windo, |CTRL-W|, etc., will consider the window as
one that can be made the "current window". A non-focusable window will be
skipped by such commands (though it can be explicitly focused by

> |nvim_set_current_win()|). Non-focusable windows are hidden from |:tabs| and
> the default tabline.

Reviewer: ```suggestion
|nvim_set_current_win()|). Non-focusable windows are not listed by |:tabs| nor
the default 'tabline'.
```
